kf_lib_data_ingest/network/utils.py

```diff
@@ -245,3 +250,58 @@ def get_open_api_v2_schema(url, entity_names,
             print(err)
 
     return output
+
+
+def get_tag_commit_sha(tag_name,
+                       tag_commit_cache_file=TAG_COMMIT_CACHE_FILE):
+    """
+    Get the commit SHA associated with the git tag `tag_name`
+
+    :param tag_name: Name of git tag (e.g. 0.3.0)
+    :type tag_name: str
+    :param tag_commit_cache_file: Path to the JSON file which caches the
+    git tag names and corresponding commit SHAs
+    :type tag_commit_cache_file: str
+    :returns commit sha associated with the supplied git tag name
+    """
+    logger.debug(f'Getting commit sha for release tag {tag_name} ...')
+
+    commit_sha = None
+    tag_commit_cache = None
+
+    # Try to get commit from tag commit cache
+    logger.debug(f'Try searching tag commit cache: {tag_commit_cache_file}')
+    if os.path.isfile(tag_commit_cache_file):
+        tag_commit_cache = read_json(tag_commit_cache_file)
+        commit_sha = tag_commit_cache.get(tag_name)
+        if commit_sha:
+            return commit_sha
+        else:
+            logger.debug(f'Could not find tag {tag_name} in tag commit cache')
+    else:
+        logger.debug('Tag commit cache does not exist yet')
+
+    # Try to cet commit from github API
```

Comment:
For completeness, I think we should try local git commands first before hitting the github api.